lib/LanguageServerCompletion/Handler/CompletionHandler.php

```diff
@@ -111,16 +111,25 @@ public function completion(CompletionParams $params, CancellationToken $token):
                         : InsertTextFormat::PLAIN_TEXT
                     ;
                 }
-                
-                $items[] = $i = CompletionItem::fromArray([
+
+                $importNameResult = $this->importClassOrFunctionName($suggestion, $params);
+
+                if ($importNameResult->isSuccessAndHasAliasedNameImport()) {
+                    // todo is this the right place?
+                    $importName = $importNameResult->getNameImport()->alias();
+                    $insertText = str_replace($name, $importName, $insertText);
+                }
+
+                $items[] = CompletionItem::fromArray([
                     'label' => $name,
                     'kind' => PhpactorToLspCompletionType::fromPhpactorType($suggestion->type()),
                     'detail' => $this->formatShortDescription($suggestion),
                     'documentation' => $suggestion->documentation(),
                     'insertText' => $insertText,
                     'sortText' => $this->sortText($suggestion),
+                    // todo convert text edit if there is an alias import
```

Comment:
The imported name and the text edit have to be changed for an alias import. I am unsure whether this belongs to the `CompletionHandler`. But I have no idea where to put it otherwise.

Reply:
Could add a bridge/adapter to avoid the coupling, I think the smell is that we import classes from CodeTransform here, you could introduce a `Model\CompletionNameImporter` interface (or whatever) and then add an adapter in `LanugageServerCompletion/Adapter/LanguageServerCodeTransform/CodeTransformCompletionNameImporter` for example.

What do you think?

Reply:
That doesn't solve the problem:
* The completion provides suggestions (e.g. `Foo`)
* The import function finds out that it has to be imported using an alias name (e.g. `BarFoo`)
* Now this has to be feed back to the completion because the imported name is now different from the original suggestion

Reply:
Why does it need to be fed back? Is there a functional problem with the code as it is?

Reply:
Because the import may result in using an alias. As described above: The original suggestion is `Foo`. Because of the import an alias has to be used. `Foo` becomes for e.g. `BarFoo`. If that happens the completion `insertText` and the `textEdit` have to be changed from `Foo` to `BarFoo`. This is easy for the `insertText` but not for the `textEdit`.

Reply:
To be concrete: this has to change if there is an alias: https://github.com/weeman1337/language-server-phpactor-extensions/blob/78b68789ff221cd084084d6e5156a0c3e071e3df/lib/LanguageServerCompletion/Handler/CompletionHandler.php#L157-L176

Reply:
Hai @dantleech . Just a ping if you have any ideas about how to solve the problem mentioned above.

Reply:
I'm sure I'm missing something: but why not refactor to pass the `$importName` in addition to `Suggestion`? The range remains constant? Only the replacement text needs to change?

Reply:
I am not sure if I am proably missing something 🙃

I had a closer look at the `TextEdit` spec. First I thought that it contains the target range. That isn't the case. I will update the code. It should work.